doc/tutorial_src/stream_test.c

```diff
@@ -94,6 +94,23 @@ Ensure(empty_paragraphs_are_ignored) {
     by_paragraph(&reader, NULL, &writer, NULL);
 }
 
+static void update_counter(void * counter) {
+    *counter = *counter + 1;
+}
+
+Ensure(using_side_effect) {
+    int number_of_times_reader_was_called = 0;
+    expect(reader, will_return('\n'));
+    always_expect(reader,
+                  will_return(EOF),
+                  with_sideeffect(&update_counter,
+                                  &number_of_times_reader_was_called));
```

Comment:
comment or document that the side effect data can't be a boxed double, so it is limited to types that will fit inside an intptr_t

Reply:
I added some test about that in the documentation